frontend/src/pages/queryBuilder/jsonEditor/JsonEditor.jsx

```diff
@@ -0,0 +1,194 @@
+import React, { useState, useContext, useEffect } from 'react';
+
+import ReactJson from 'react-json-view';
+import IconButton from '@material-ui/core/IconButton';
+import Dialog from '@material-ui/core/Dialog';
+import DialogTitle from '@material-ui/core/DialogTitle';
+import DialogContent from '@material-ui/core/DialogContent';
+import DialogActions from '@material-ui/core/DialogActions';
+
+import CloudUploadIcon from '@material-ui/icons/CloudUpload';
+import CheckIcon from '@material-ui/icons/Check';
+// import UndoIcon from '@material-ui/icons/Undo';
+import CloseIcon from '@material-ui/icons/Close';
+
+import trapiUtils from '~/utils/trapi';
+import usePageStatus from '~/stores/usePageStatus';
+import AlertContext from '~/context/alert';
+import QueryBuilderContext from '~/context/queryBuilder';
+
+/**
+ * Query Builder json editor interface
+ * @param {bool} show whether to show the json editor or not
```

Comment:
I think we decided that these docstrings should have colons?
```suggestion
 * @param {bool} show: whether to show the json editor or not
```

Reply:
Colons make the hover docs look weird in VSCode, so we're going with dashes. Added in e2642a842f621a68b21266281add5fa5b5501b74